janusgraph-core/src/main/java/org/janusgraph/graphdb/database/IndexSerializer.java

```diff
@@ -105,24 +105,24 @@ public static void register(final MixedIndexType index, final PropertyKey key, f
 
     }
 
-//    public boolean supports(final String indexName, final Class<?> dataType, final Parameter[] parameters) {
-//        IndexInformation indexinfo = indexes.get(indexName);
-//        Preconditions.checkArgument(indexinfo != null, "Index is unknown or not configured: %s", indexName);
-//        return indexinfo.supports(new StandardKeyInformation(dataType,parameters));
-//    }
-
     public boolean supports(final MixedIndexType index, final ParameterIndexField field) {
-        IndexInformation indexinfo = mixedIndexes.get(index.getBackingIndexName());
+        final IndexInformation indexinfo = mixedIndexes.get(index.getBackingIndexName());
         Preconditions.checkArgument(indexinfo != null, "Index is unknown or not configured: %s", index.getBackingIndexName());
         return indexinfo.supports(getKeyInformation(field));
     }
 
     public boolean supports(final MixedIndexType index, final ParameterIndexField field, final JanusGraphPredicate predicate) {
-        IndexInformation indexinfo = mixedIndexes.get(index.getBackingIndexName());
+        final IndexInformation indexinfo = mixedIndexes.get(index.getBackingIndexName());
         Preconditions.checkArgument(indexinfo != null, "Index is unknown or not configured: %s", index.getBackingIndexName());
         return indexinfo.supports(getKeyInformation(field),predicate);
     }
 
+    public IndexFeatures features(final MixedIndexType index) {
+        final IndexInformation indexinfo = mixedIndexes.get(index.getBackingIndexName());
+        Preconditions.checkArgument(indexinfo != null, "Index is unknown or not configured: %s", index.getBackingIndexName());
```

Comment:
minor just concatenate (+) the string and backingIndexName. Was discussed elsewhere in jg with @sjudeng that concatenation is faster than printf-stlye expressions.